src/TidePublicationOperation.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace Drupal\tide_publication;
+
+use Drupal\Core\Config\Config;
+use Drupal\field\Entity\FieldConfig;
+use Drupal\taxonomy\Entity\Vocabulary;
+
+/**
+ * Tide publication modules operations.
+ */
+class TidePublicationOperation {
+
+  /**
+   * The bundle name.
+   */
+  const BUNDLE = 'publication';
+
+  /**
+   * The entity reference field type.
+   */
+  const ENTITY_REFERENCE_FIELD_TYPE = 'entity_reference';
+
+  /**
+   * The publication authors field name.
+   */
+  const PUBLICATION_AUTHORS_FIELD_NAME = 'field_publication_authors';
+
+  /**
+   * The publication authors field description.
+   */
+  const PUBLICATION_AUTHORS_FIELD_DESCRIPTION = 'Start typing to choose one publication author.';
+
+  /**
+   * The publication authors field storage.
+   */
+  const PUBLICATION_AUTHORS_FIELD_STORAGE = 'field.storage.node.field_publication_authors';
+
+  /**
+   * Update field_publication_authors to use new vocabulary.
+   *
+   * @param \Drupal\Core\Config\Config $config
+   *   The config object.
+   * @param string $vocabulary
+   *   The new vocabulary.
+   */
+  public function updateFieldPublicationAuthorsVocabulary(Config $config, string $vocabulary) {
```

Comment:
why are we separating this like a helper function ? This needs to be used only once, after that this code will just sit here. Better to put it in update hook, cause after that there is no use of it. Also the values you set are constant, that also means this function can not be used for any other fields.

Reply:
Hey @MdNadimHossain , I am following Matthew's suggestions from this [pr](https://github.com/dpc-sdp/tide_core/pull/417#discussion_r1296544317) 
@krakerag, @MdNadimHossain  let me know guys how should I proceed then.

Reply:
I don't think this applies here, cause this function not going to be used after updating this field. Also for fresh install in the new CMS the update the updates will get in from the config/install for these fields. If there are scope where you can reuse this function then it make sense to separate it in a class. But this one not needed in the install hook and no other use case of reusing it as well.
CC: @krakerag

Reply:
This is an interesting one. All update hook code is throwaway code, but we established a pattern in tide_search and tide_core to minimize the code in the .install file, for the preference of putting code in a Helper or Operation class that lives in the src folder.

I think @yeniatencio is just following that established pattern here.

That said, I am happy to see either - I don't feel like being strict on install file rules is going to gain anything from a code readability standpoint.

I would be more strict on the createVocabulary method being in a reusable code location which is good on L86 below.